### DISCUSSION TITLE
See progress with batch processing on a large collection?

### DISCUSSION THREAD

**Discussion by suzanv** [explosion/spaCy#9113] [Help: Coding & Implementations] [answered]
(If this was asked before, please refer me to that location)

I am batch processing Spacy for document similarity. My collection contains 4 million short texts.
```
spacy_nlp = spacy.load('en_core_web_lg')
docs = list(spacy_nlp.pipe(texts,disable=["tagger", "parser", "lemmatizer", "ner","attribute_ruler"], batch_size=1000))
```

Since this runs for a long time, I would like to see progress. Is there a way to get logging output from nlp.pipe? E.g. the batch number it is currently processing?

**Comment by polm** [comment, marked as answer]
There is not a builtin flag or something for this, but you can use [tqdm][] or other generic progress trackers.

[tqdm]: https://github.com/tqdm/tqdm

```
from tqdm import tqdm

nlp = spacy.load(...)
texts = ...
docgen = nlp.pipe(texts, ...)
docs = list(tqdm(docgen, total=len(texts)))
```

**Reply by suzanv** [reply]
Thanks, this is a good solution!
